### PR TITLE
Fix error catch

### PR DESCRIPTION
**Pull request by tescher**
FB is returning a response <> 200, but the catch and logging are failing because "error" is null. Need to fix this so I can figure out what's going on with the FB API.

### CODE REVIEW

File: routes/index.js [hunk 1]
```diff
@@ -304,8 +304,12 @@ function sendFBMessage(recipientId, messageText) {
 
     }, function (error, response, body) {
         if (error || (response.statusCode != 200)) {
-            logger.error("Failed calling Send API: ", error.message);
-            logger.error("Failed calling Send API", response.statusCode, response.statusMessage);
+            if (error) {
+                logger.error("Failed calling Send API: ", error.message);
+            }
+            if (response) {
+                logger.error("Failed calling Send API", response.statusCode, response.statusMessage);
```
Comment (NigelKibodeaux) [inline routes/index.js]: looks good except that you'll need to call this a little differently. See here: https://github.com/codeforanchorage/realtime-bus-sms/blob/c1ace6ca9c5a52df5872ddde5d07bf9f9b2bb9ca/lib/logger.js#L45

**Comment by tescher** [comment]
Changed.

Tim Escher
tim@eschers.com
608-432-5799

**Comment by redcap3000** [comment]
Hi... was bored and saw this because I've been dealing with this a lot. Null can sometimes can be mistaken for false, so some circumstances may report a false error.. So best way to handle every case, in the event the response changes in the future (or other internet weirdness), is to check for typeof null in addition null.
```
if(typeof error == 'null' || !error || error == null){
   // no error
   if(typeof response == null || !response || response == null){
     // no response, but no error
   }else{
      // response
   }
}else{
   // error
}
```

**Comment by NigelKibodeaux** [comment]
In this case we don't need to differentiate between null, false, and undefined. `!error` covers all those. 

Also I don't think there is anything in javascript where typeof is 'null'. Even null itself has a type of 'Object' (unfortunately).

**Comment by redcap3000** [comment]
Consider this: 
http://stackoverflow.com/questions/27632391/why-null-false-and-null-true-both-return-false

It can be problematic to use boolean comparative operators when null is not a boolean type.

**Pull request merged**